# Computational Biology Software

This software is designed for computational biology tasks, utilizing Python for programming and Strealtit for the user interface. It includes implementations of algorithms such as the Smith-Watermann Algorithm and the Needleman-Wunsch Algorithm. The Smith-Watermann Algorithm is used for determining similar regions between two nucleic acid or protein sequences. On the other hand, the Needleman-Wunsch Algorithm is employed for both global best and local best alignments. Additionally, the software incorporates the UPGMA algorithm, which stands for the unweighted pair group method with arithmetic mean. UPGMA is a clustering technique that constructs phylogenetic trees based on distance matrix data. It assumes a constant rate of substitution over time and phylogenetic lineages, although this assumption is not always accurate. UPGMA finds applications in various fields such as ecology, systematics, numerical taxonomy, and distance-matrix methods.

### Links to Applications:

##### Needleman Wunsch Software
https://computational-biology-projects-needlemanwunschalgorithm.streamlit.app/ [link](https://computational-biology-projects-needlemanwunschalgorithm.streamlit.app/)

#### Smith Waterman Software
https://computational-biology-projects-psasmithwaterman.streamlit.app/
[link](https://computational-biology-projects-psasmithwaterman.streamlit.app/)

#### UPGMA Software
https://computational-biology-projects-upgmaphylotree.streamlit.app/
[link](https://computational-biology-projects-upgmaphylotree.streamlit.app/)
